src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -10,12 +10,11 @@ with `n` vertices.
 function CompleteGraph(n::T) where {T <: Integer}
     n <= 0 && return SimpleGraph{T}(0)
     ne = Int(n * (n - 1) ÷ 2)
-
     fadjlist = Vector{Vector{T}}(undef, n)
     @inbounds @simd for u = 1:n
         listu = Vector{T}(undef, n-1)
-        listu[1:(u-1)] = UnitRange{T}(1, u - 1)
-        listu[u:(n-1)] = UnitRange{T}(u + 1, n)
+        listu[1:(u-1)] = 1:(u - 1)
+        listu[u:(n-1)] = (u + 1):n
         fadjlist[u] = listu
```

Comment:
Getting back to this - why does it not matter that we preserve the eltype here?

Reply:
This was an error that I made in the previous pr and the reason why I started this pr. I though that you can have code such as
```
    UnitRange{Int8}(1, 5)
```
and then Julia would automatically convert 1 and 5 to `Int8` but instead you apparently get an error. This unfortunately was never caught by the tests. 
When we assign the values of an `UnitRange{Int64}` to `Vector{Int8}[:]`, Julia will do the conversion to `Int8` so the new solution is correct.

Reply:
Actually, what I wrote is not completely true - `UnitRange{Int8}(1, 5)` is totally valid but if the arguments are of mixed integer type, such as `UnitRange{Int8}(1, Int8(5)}`, then it will error.